### ISSUE TITLE
[FEATURE] Ability to suppress run-time null checks of @Getter and @Setter

### ISSUE THREAD

**Issue by nlwillia** [rzwitserloot/lombok#2078]
The [changelog](https://projectlombok.org/changelog) for v1.18.4 contains a surprising change to the behavior of `@Getter` and `@Setter` when a nullability annotation is used.

Lombok has always had its own `@NonNull` annotation which injected runtime null checking, but you didn't _have_ to use it.  If you wanted to use a documentation annotation like JetBrains [@NotNull](https://github.com/JetBrains/java-annotations/blob/master/java8/src/main/java/org/jetbrains/annotations/NotNull.java), you could.  Lombok would copy it to the getters and setters, but it wouldn't inject any runtime checks.

As of 1.18.4, there doesn't appear to be any way of avoiding runtime checks.  I generally have IntelliJ IDEA configured to [compile runtime null checks](https://www.jetbrains.com/help/idea/compiler.html) for development but omit them from server builds.  Lombok's interjection here is redundant with what my IDE is already doing and unwanted on the server.

I tried `lombok.nonNull.flagUsage` values of `warning` and `allow`, but neither kept the runtime check from being added.  (I'm not using "NonNull", I'm using JetBrains "No**t**Null".)  The new `lombok.copyableAnnotations` property seems to be add-only.  I tried `= []` and it didn't change anything.

In a large code base where nullability annotations are already being used but only for documentation purposes, Lombok's new behavior poses an unwanted risk.  It's now possible that a usage pattern that worked (even if it was ill-advised) could start to actively fail.  There should be a way to configure the previous behavior.

In other words, I want to configure `public @NotNull @Getter @Setter String foo;` (or the `@Data` versions of the same) to produce **only** the following:
```
public @NotNull String getFoo() {
    return foo;
}
public void setFoo(@NotNull String foo) {
    this.foo = foo;
}
```

**Comment by rspilker** [comment]
Short term work-around: You can create your own annotation, and teach IntelliJ to treat is like a NotNull. Since lombok does not know your annotation, we will not generate a null-check.

We are considering a long term solution.

The list of built-in non-null annotations is not modifiable from the configuration system. They are hard-coded.

Regarding the _syntax_ of the configuration system, the way to make remove individual entries from a list is `lombok.copyableAnnotations -= my.Annotation`, and to remove all of them by `clear lombok.copyableAnnotations`.

**Comment by rspilker** [comment]
I think we now have a solution for you.

In `lombok.config`, you can now configure `lombok.nonNull.exceptionType = Assertion`.

If you do so, you will get `assert foo != null : "foo is marked non-null but is null"`.

Assuming that you will not run the application with `-ea`, this means you will not get any errors in production.

**Comment by nlwillia** [comment]
That sounds like a reasonable compromise.  Thanks.

**Comment by rspilker** [comment]
Thanks for the feedback.

**Comment by nlwillia** [comment]
Confirmed working in the edge release.
